contracts/multisig/test/asset_proxy_owner.ts

```diff
@@ -1,511 +1,641 @@
+                data,
+                destinations,
+                signerAddresses,
+                newTimeLock.toNumber(),
+                { values },
+            );
+            assertReceiverCalledFromLogs(results.executionTxReceipt.logs, data, destinations, results.txId, values);
+        });
+        it('should be able to call multiple functions with a single destination and no values', async () => {
+            const data = [hexRandom(), hexRandom()];
+            const destinations = [receiver.address, receiver.address];
+            const results = await assetProxyOwnerWrapper.submitConfirmAndExecuteTransactionAsync(
+                data,
+                destinations,
+                signerAddresses,
+                DEFAULT_TIME_LOCK,
+            );
+            assertReceiverCalledFromLogs(results.executionTxReceipt.logs, data, destinations, results.txId);
+        });
+        it('should be able to call multiple functions with different destinations and values', async () => {
+            const receiver2 = await ContractCallReceiverContract.deployFrom0xArtifactAsync(
+                artifacts.ContractCallReceiver,
+                env.provider,
+                env.txDefaults,
+                {},
+            );
+            const data = [hexRandom(), hexRandom()];
+            const destinations = [receiver.address, receiver2.address];
+            const values = [INITIAL_BALANCE.dividedToIntegerBy(4), INITIAL_BALANCE.dividedToIntegerBy(3)];
+            const results = await assetProxyOwnerWrapper.submitConfirmAndExecuteTransactionAsync(
+                data,
+                destinations,
+                signerAddresses,
+                DEFAULT_TIME_LOCK,
+                { values },
+            );
+            assertReceiverCalledFromLogs(results.executionTxReceipt.logs, data, destinations, results.txId, values);
+        });
+        it('should be able to call a combination of registered and unregistered functions', async () => {
+            const data = [hexRandom(), hexRandom()];
+            const destinations = [receiver.address, receiver.address];
+            const newTimeLock = new BigNumber(DEFAULT_TIME_LOCK).dividedToIntegerBy(2);
+            await assetProxyOwner.registerFunctionCallBypassWallet.awaitTransactionSuccessAsync(
+                true,
+                data[0].slice(0, 10),
+                receiver.address,
+                newTimeLock,
+            );
+            const results = await assetProxyOwnerWrapper.submitConfirmAndExecuteTransactionAsync(
+                data,
+                destinations,
+                signerAddresses,
+                DEFAULT_TIME_LOCK,
+            );
+            assertReceiverCalledFromLogs(results.executionTxReceipt.logs, data, destinations, results.txId);
+        });
+        it('should fail if a single function has not passed the timelock', async () => {
```

Comment:
If you make the change that I recommended regarding reverting if any of the functions have not elapsed their timelock, this unit test can be made even stronger by requiring that no `.call` operations were made. This should be an easy change since you are using the receiver pattern :)